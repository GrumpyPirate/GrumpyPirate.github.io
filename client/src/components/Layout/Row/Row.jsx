import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import classes from './Row.scss';

const Row = ({ align, justify, children }) => (
  <div
    className={classnames(
      classes.row, {
        [classes[`row--align-${align}`]]: align,
        [classes[`row--justify-${justify}`]]: justify,
      },
    )}
  >
    {children}
  </div>
);

Row.propTypes = {
  align: PropTypes.oneOf(['top', 'middle', 'bottom']),
  justify: PropTypes.oneOf(['left', 'center', 'right', 'space-between', 'space-around']),
  children: PropTypes.node,
};

Row.defaultProps = {
  align: null,
  justify: null,
  children: null,
};

export default Row;
