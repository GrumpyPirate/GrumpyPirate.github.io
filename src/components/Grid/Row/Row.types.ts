export type RowXAlign = 'left' | 'center' | 'right';
export type RowYAlign = 'top' | 'center' | 'bottom' | 'stretch';

export interface RowProps {
  xAlign?: RowXAlign;
  yAlign?: RowYAlign;
}
