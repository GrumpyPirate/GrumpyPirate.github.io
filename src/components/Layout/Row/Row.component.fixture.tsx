import React from 'react';

import Column from 'components/Layout/Column/Column.component';

export default {
  children: (
    <Column>
      <div>Test</div>
    </Column>
  ),
};
