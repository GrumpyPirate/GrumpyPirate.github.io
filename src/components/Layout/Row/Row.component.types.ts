import { ChildrenProps } from 'types/common';

export interface RowProps extends ChildrenProps {
  align?: 'top' | 'middle' | 'bottom';
  alignSM?: 'top' | 'middle' | 'bottom';
  alignMD?: 'top' | 'middle' | 'bottom';
  alignLG?: 'top' | 'middle' | 'bottom';
  alignXL?: 'top' | 'middle' | 'bottom';
  justify?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifySM?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyMD?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyLG?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
  justifyXL?: 'left' | 'center' | 'right' | 'space-between' | 'space-around';
}
