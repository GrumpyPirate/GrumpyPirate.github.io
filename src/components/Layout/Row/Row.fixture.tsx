import React from 'react';

import Column from 'components/Layout/Column/Column';

export default {
  children: (
    <Column>
      <div>Test</div>
    </Column>
  ),
};
