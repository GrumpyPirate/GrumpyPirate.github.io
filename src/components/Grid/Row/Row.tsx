import styled, { css } from 'styled-components';

import { gutterWidths, mediaQueries, rem } from 'styles';

import { RowProps } from './Row.types';

const Row = styled.div<RowProps>`
  display: flex;
  flex-flow: row wrap;
  margin-right: ${rem(gutterWidths.xs / -2)};
  margin-left: ${rem(gutterWidths.xs / -2)};

  ${({ xAlign }) => {
    switch (xAlign) {
      case 'center':
        return css`
          justify-content: center;
        `;
      case 'right':
        return css`
          justify-content: flex-end;
        `;
      default:
        return '';
    }
  }}

  ${({ yAlign }) => {
    switch (yAlign) {
      case 'top':
        return css`
          align-items: flex-start;
        `;
      case 'center':
        return css`
          align-items: center;
        `;
      case 'bottom':
        return css`
          align-items: flex-end;
        `;
      case 'stretch':
        return css`
          align-items: stretch;
        `;
      default:
        return '';
    }
  }}

  @media ${mediaQueries.md} {
    margin-right: ${rem(gutterWidths.md / -2)};
    margin-left: ${rem(gutterWidths.md / -2)};
  }
`;

export default Row;
