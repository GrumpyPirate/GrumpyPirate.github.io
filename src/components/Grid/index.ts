import Column from './Column/Column';
import Container from './Container/Container';
import Row from './Row/Row';

export { Container, Row, Column };
