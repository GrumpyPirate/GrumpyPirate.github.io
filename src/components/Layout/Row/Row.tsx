import classnames from 'classnames';
import React, { FunctionComponent } from 'react';

import classes from './Row.scss';
import { RowProps } from './Row.types';

const Row: FunctionComponent<RowProps> = ({
  align,
  alignSM,
  alignMD,
  alignLG,
  alignXL,
  justify,
  justifySM,
  justifyMD,
  justifyLG,
  justifyXL,
  children,
}) => (
  <div
    className={classnames(classes.row, {
      [classes[`row--align-${align}`]]: align,
      [classes[`row--align-sm-${alignSM}`]]: alignSM,
      [classes[`row--align-md-${alignMD}`]]: alignMD,
      [classes[`row--align-lg-${alignLG}`]]: alignLG,
      [classes[`row--align-xl-${alignXL}`]]: alignXL,
      [classes[`row--justify-${justify}`]]: justify,
      [classes[`row--justify-sm-${justifySM}`]]: justifySM,
      [classes[`row--justify-md-${justifyMD}`]]: justifyMD,
      [classes[`row--justify-lg-${justifyLG}`]]: justifyLG,
      [classes[`row--justify-xl-${justifyXL}`]]: justifyXL,
    })}
  >
    {children}
  </div>
);

export default Row;
