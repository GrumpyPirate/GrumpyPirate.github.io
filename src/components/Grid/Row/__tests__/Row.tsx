import { render } from '@testing-library/react';
import React from 'react';

import { Column } from 'components/Grid';

import Row from '../Row';

describe('Components/Grid/Row', () => {
  const props = {
    children: (
      <>
        <Column>Column 1</Column>
        <Column>Column 2</Column>
        <Column>Column 3</Column>
      </>
    ),
  };

  test('should render as expected, without crashing', () => {
    const { container } = render(<Row {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('xAlign', () => {
    test('should honour an xAlign of "center"', () => {
      const { container } = render(<Row {...props} xAlign="center" />);

      expect(container.firstChild).toHaveStyleRule('justify-content', 'center');
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should honour an xAlign of "right"', () => {
      const { container } = render(<Row {...props} xAlign="right" />);

      expect(container.firstChild).toHaveStyleRule('justify-content', 'flex-end');
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('yAlign', () => {
    test('should honour a yAlign of "top"', () => {
      const { container } = render(<Row {...props} yAlign="top" />);

      expect(container.firstChild).toHaveStyleRule('align-items', 'flex-start');
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should honour a yAlign of "center"', () => {
      const { container } = render(<Row {...props} yAlign="center" />);

      expect(container.firstChild).toHaveStyleRule('align-items', 'center');
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should honour a yAlign of "bottom"', () => {
      const { container } = render(<Row {...props} yAlign="bottom" />);

      expect(container.firstChild).toHaveStyleRule('align-items', 'flex-end');
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should honour a yAlign of "stretch"', () => {
      const { container } = render(<Row {...props} yAlign="stretch" />);

      expect(container.firstChild).toHaveStyleRule('align-items', 'stretch');
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
