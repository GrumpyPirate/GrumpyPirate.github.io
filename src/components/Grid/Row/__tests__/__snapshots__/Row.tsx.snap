// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Grid/Row should render as expected, without crashing 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row xAlign should honour an xAlign of "center" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row xAlign should honour an xAlign of "right" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row yAlign should honour a yAlign of "bottom" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row yAlign should honour a yAlign of "center" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row yAlign should honour a yAlign of "stretch" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;

exports[`Components/Grid/Row yAlign should honour a yAlign of "top" 1`] = `
.c1 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Column 1
  </div>
  <div
    class="c1"
  >
    Column 2
  </div>
  <div
    class="c1"
  >
    Column 3
  </div>
</div>
`;
